agent/lib/kontena/workers/weave_worker.rb

```diff
@@ -72,7 +72,7 @@ def weave_attach(container)
     # @param [String] container_id
     # @param [String] overlay_cidr
     def attach_overlay(container_id, overlay_cidr)
-      Actor[:network_adapter].exec(['--local', 'attach', overlay_cidr, '--rewrite-hosts', container.id])
+      Actor[:network_adapter].exec(['--local', 'attach', overlay_cidr, '--rewrite-hosts', container_id])
```

Comment:
For testing, it might be better to have something like this?

```ruby
   def network_adapter
      Actor[:network_adapter]
   end
   def attach_overlay(container)
       network_adapter.attach_container(container.id, container.overlay_cidr)
   end
```

And then spec with 

```
    subject do
        described_class.new
    end
    let :network_adapter
        instance_double(Kontena::NetworkAdapters::Weave)
    end
    before
       allow(subject).to receive(:network_adapter).and_return(network_adapter)
    end
    it '...' do
        expect(network_adapter).to receive(:attach_container).with('aaaaabbbcc', '10.81.128.0/16')
    end
```